### PR TITLE
[Range slider] Center thumbs with track click

### PR DESCRIPTION
**Pull request by solonaarmstrong**
### WHY are these changes introduced?

Follow-up to https://github.com/Shopify/polaris-react/pull/987#issuecomment-461999835 Part of https://github.com/Shopify/polaris-react/pull/784

### WHAT is this pull request doing?

Aligns the click position with the center of the thumbs, rather than the left edge.

#### Before

range-slider-off

#### After

range-slider-center

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import * as React from 'react';
import {Page, RangeSlider, Heading, Stack, TextField} from '../src';
import {Key} from '../src/types';

interface State {
  intermediateTextFieldValue: [number, number];
  dualValue: [number, number];
  singleValue: number;
}

const initialValue: [number, number] = [500, 1000];

export default class Playground extends React.Component<{}, State> {
  state: State = {
    intermediateTextFieldValue: initialValue,
    dualValue: initialValue,
    singleValue: 50,
  };

  handleRangeSliderChange = (value) => {
    this.setState({dualValue: value, intermediateTextFieldValue: value});
  };

  handleLowerTextFieldChange = (value) => {
    const upperValue = this.state.dualValue[1];
    this.setState({intermediateTextFieldValue: [parseInt(value), upperValue]});
  };

  handleUpperTextFieldChange = (value) => {
    const lowerValue = this.state.dualValue[0];
    this.setState({intermediateTextFieldValue: [lowerValue, parseInt(value)]});
  };

  handleLowerTextFieldBlur = () => {
    const upperValue = this.state.dualValue[1];
    const value = this.state.intermediateTextFieldValue[0];
    this.setState({dualValue: [Number(value), upperValue]});
  };

  handleUpperTextFieldBlur = () => {
    const lowerValue = this.state.dualValue[0];
    const value = this.state.intermediateTextFieldValue[1];
    this.setState({dualValue: [lowerValue, Number(value)]});
  };

  handleEnterKeyPress = (event) => {
    const newValue = this.state.intermediateTextFieldValue;
    const oldValue = this.state.dualValue;
    if (event.keyCode === Key.Enter && newValue !== oldValue) {
      this.setState({dualValue: newValue});
    }
  };

  handleSingleChange = (value: number) => {
    this.setState({singleValue: value});
  };

  render() {
    const {dualValue, intermediateTextFieldValue} = this.state;
    const prefix = '$';
    const min = 0;
    const max = 2000;
    const step = 1;
    const disabled = false;
    const lowerTextFieldValue =
      intermediateTextFieldValue[0] === dualValue[0]
        ? dualValue[0]
        : intermediateTextFieldValue[0];
    const upperTextFieldValue =
      intermediateTextFieldValue[1] === dualValue[1]
        ? dualValue[1]
        : intermediateTextFieldValue[1];

    return (
      <Page title="Playground">
        <div style={{marginTop: '20px'}} onKeyDown={this.handleEnterKeyPress}>
          <RangeSlider
            label="Money spent is between"
            value={dualValue}
            prefix={prefix}
            output
            min={min}
            max={max}
            step={step}
            disabled={disabled}
            onChange={this.handleRangeSliderChange}
          />
          <Stack distribution="equalSpacing" spacing="extraLoose">
            <TextField
              label="Min money spent"
              type="number"
              value={`${lowerTextFieldValue}`}
              prefix={prefix}
              min={min}
              max={max}
              step={step}
              disabled={disabled}
              onChange={this.handleLowerTextFieldChange}
              onBlur={this.handleLowerTextFieldBlur}
            />
            <TextField
              label="Max money spent"
              type="number"
              value={`${upperTextFieldValue}`}
              prefix={prefix}
              min={min}
              max={max}
              step={step}
              disabled={disabled}
              onChange={this.handleUpperTextFieldChange}
              onBlur={this.handleUpperTextFieldBlur}
            />
          </Stack>
        </div>

        <div style={{width: '300px', height: '100px', marginTop: '100px'}}>
          <Heading>Native range slider</Heading>
          <br />
          <RangeSlider
            label=""
            labelHidden
            value={this.state.singleValue}
            onChange={this.handleSingleChange}
            output
            min={5}
            max={30}
            prefix="low"
            suffix="high"
          />
        </div>
      </Page>
    );
  }
}

```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)

### CODE REVIEW

File: src/components/RangeSlider/components/DualThumb/DualThumb.tsx [hunk 1]
```diff
@@ -306,6 +306,25 @@ export default class DualThumb extends React.Component<Props, State> {
...
+
+    if (distanceFromLowerThumb <= distanceFromUpperThumb) {
+      this.setValue([centeredXPosition, value[1]], Control.Upper);
+      registerMouseMoveHandler(this.handleMouseMoveThumbLower);
+    } else {
+      this.setValue([value[0], centeredXPosition], Control.Lower);
+      registerMouseMoveHandler(this.handleMouseMoveThumbUpper);
+    }
+  }
+
```
Comment (solonaarmstrong) [inline src/components/RangeSlider/components/DualThumb/DualThumb.tsx]: I moved the method up to sit next to the other mouse event methods.

Comment (tmlayton) [reply src/components/RangeSlider/components/DualThumb/DualThumb.tsx]: Not too big of a deal here, but would avoid moving code around which is also changing. At a minimum it is hard for reviewers to tell what has actually changed and worst case potentially causes reviewers to miss important parts of a PR and result in shipping bad code.

**Comment by solonaarmstrong** [comment]
I left the tests failing because there's no sense in changing them if this isn't the right fix.

**Comment by tmlayton** [comment]
Closing in lieu of #1440

**Pull request closed without merging**